packages/react-router-dev/typegen.ts

```diff
@@ -25,20 +27,34 @@ function getDirectory(ctx: Context) {
   return Path.join(ctx.rootDirectory, ".react-router/types");
 }
 
-export function getPath(ctx: Context, route: RouteManifestEntry): string {
-  return Path.join(
-    getDirectory(ctx),
-    Path.relative(ctx.rootDirectory, ctx.appDirectory),
-    Path.dirname(route.file),
-    "+types." + Pathe.filename(route.file) + ".d.ts"
-  );
-}
+export async function watch(
+  rootDirectory: string,
+  options: { configFile?: string } = {}
+) {
+  const watchStart = performance.now();
 
-export async function watch(rootDirectory: string) {
-  const vitePluginCtx = await loadPluginContext({ root: rootDirectory });
-  const routesTsPath = Path.join(
+  const vitePluginCtx = await loadPluginContext({
+    root: rootDirectory,
+    configFile: options.configFile,
+  });
+  const routeConfigFile = findEntry(
```

Comment:
Not blocking, but I wonder if the resolved config file path should be on the context object?

Reply:
Yes, I was thinking the same thing as I worked on this.

Though there's a bigger refactor we should do too; right now, `loadPluginContext` throws its own errors and calls `process.exit`, but I want finer grain control when running in watch mode. I think a nice solution would be to refactor `loadPluginContext` (and any code it depends on) to return a `Result` type with possible errors rather than throw/exit.